scipy/stats/_continuous_distns.py

```diff
@@ -1611,6 +1611,19 @@ def _sf(self, x, a):
                         0.5*sc.gammaincc(a, x),
                         0.5 + 0.5*sc.gammainc(a, -x))
 
+    def _entropy(self, a):
+        a = np.asarray([a])
+        def h1(a):
+            h1 = a + np.log(2) + sc.gammaln(a) + (1 - a) * sc.digamma(a)
+            return h1
+
+        def h2(a):
+            h2 = np.log(2) + 0.5 * (1 + np.log(a) + np.log(2 * np.pi))
+            return h2
+
+        h = _lazywhere(a > 1e13, (a), f=h2, f2=h1)
```

Comment:
```suggestion
        h = _lazywhere(a > 1e10, (a), f=h2, f2=h1)
```

Turns out problems start way before `1e13`:

![Dgamma_Entropy_large_a_problems](https://user-images.githubusercontent.com/40656107/219783060-8873a67f-0071-4a96-9d6f-82136d6af002.png)

Zoomed in:

![Dgamma_Entropy_large_a_problems_zoom](https://user-images.githubusercontent.com/40656107/219783104-74696f68-609a-4f6e-ae3b-1dd8b22bb645.png)

With a lower cutoff, we get around this.